src/main/java/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.kt

```diff
@@ -0,0 +1,710 @@
+import net.atos.client.zgw.zrc.ZRCClientService
+import net.atos.client.zgw.zrc.model.Zaak
+import net.atos.client.zgw.zrc.model.ZaakInformatieobject
+import net.atos.client.zgw.ztc.ZTCClientService
+import net.atos.client.zgw.ztc.util.InformatieObjectTypeUtil
+import net.atos.zac.app.audit.converter.RESTHistorieRegelConverter
+import net.atos.zac.app.audit.model.RESTHistorieRegel
+import net.atos.zac.app.informatieobjecten.converter.RESTInformatieobjectConverter
+import net.atos.zac.app.informatieobjecten.converter.RESTInformatieobjecttypeConverter
+import net.atos.zac.app.informatieobjecten.converter.RESTZaakInformatieobjectConverter
+import net.atos.zac.app.informatieobjecten.model.RESTDocumentCreatieGegevens
+import net.atos.zac.app.informatieobjecten.model.RESTDocumentCreatieResponse
+import net.atos.zac.app.informatieobjecten.model.RESTDocumentVerplaatsGegevens
+import net.atos.zac.app.informatieobjecten.model.RESTDocumentVerwijderenGegevens
+import net.atos.zac.app.informatieobjecten.model.RESTDocumentVerzendGegevens
+import net.atos.zac.app.informatieobjecten.model.RESTEnkelvoudigInformatieObjectVersieGegevens
+import net.atos.zac.app.informatieobjecten.model.RESTEnkelvoudigInformatieobject
+import net.atos.zac.app.informatieobjecten.model.RESTGekoppeldeZaakEnkelvoudigInformatieObject
+import net.atos.zac.app.informatieobjecten.model.RESTInformatieobjectZoekParameters
+import net.atos.zac.app.informatieobjecten.model.RESTInformatieobjecttype
+import net.atos.zac.app.informatieobjecten.model.RESTZaakInformatieobject
+import net.atos.zac.app.zaken.converter.RESTGerelateerdeZaakConverter
+import net.atos.zac.app.zaken.model.RelatieType
+import net.atos.zac.authentication.LoggedInUser
+import net.atos.zac.configuratie.ConfiguratieService
+import net.atos.zac.documentcreatie.DocumentCreatieService
+import net.atos.zac.documentcreatie.model.DocumentCreatieGegevens
+import net.atos.zac.documenten.InboxDocumentenService
+import net.atos.zac.documenten.OntkoppeldeDocumentenService
+import net.atos.zac.enkelvoudiginformatieobject.EnkelvoudigInformatieObjectLockService
+import net.atos.zac.event.EventingService
+import net.atos.zac.flowable.FlowableTaskService
+import net.atos.zac.flowable.TaakVariabelenService
+import net.atos.zac.policy.PolicyService
+import net.atos.zac.util.UriUtil
+import net.atos.zac.webdav.WebdavHelper
+import net.atos.zac.websocket.event.ScreenEventType
+import org.apache.commons.lang3.BooleanUtils
+import org.apache.commons.lang3.StringUtils
+import org.jboss.resteasy.annotations.providers.multipart.MultipartForm
+import java.io.IOException
+import java.net.URI
+import java.util.UUID
+
+@Singleton
+@Path("informatieobjecten")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+@Suppress("TooManyFunctions", "LargeClass", "LongParameterList", "TooGenericExceptionThrown")
+class InformatieObjectenRESTService @Inject constructor(
+    private val drcClientService: DRCClientService,
```

Comment:
I see we need to suppress many warnings "TooManyFunctions", "LargeClass", "LongParameterList", "TooGenericExceptionThrown" - sure indication this should really be split into smaller "specific" services...